docs/07-theme-developer-tutorial/06-js-api.md

```diff
@@ -0,0 +1,110 @@
+  };
+  api.headerIcons.add(
+    "my-unique-icon-name",
+    <template>
+      <li>
+        <DButton
+          @action={{sayHello}}
+          @icon="wand-magic"
+          class="icon btn-flat"
+        />
+      </li>
+    </template>,
+    { before: "search" }
+  );
+});
+```
+
+### replaceIcon()
+
+```js
+api.replaceIcon(source, destination);
+```
+
+With this method, you can easily replace any Discourse icon with another. For example, we have [a theme component](https://meta.discourse.org/t/change-the-like-icon/87748) that replaces the heart icon for like with a thumbs-up icon
+
+### decorateCookedElement()
+
+`api.decorateCookedElement()` allows you to customize the rendered content of Discourse posts. This can be used for anything from simple formatting changes, all the way up to advanced integrated UIs like the builtin 'poll' plugin.
+
+The API should be passed a callback function which will be run for every post when it's rendered to the screen. The first argument to the callback will be the post's root HTML element, and the second will be a helper.
+
+A simple example which appends content to every post would look like:
+
+```js
+api.decorateCookedElement((element, helper) => {
+  const myNewParagraph = document.createElement("p");
+  myNewParagraph.textContent = "Hello, this is appended to every post!";
+  element.appendChild(myNewParagraph);
+});
+```
+
+Or for a more advanced UI, you can render a glimmer component into a post. For example, to render the counter component we authored earlier into every post, you could do something like this:
+
+```js
+import { apiInitializer } from "discourse/lib/api";
+import WelcomeBanner from "../components/welcome-banner";
+
+export default apiInitializer((api) => {
+  api.decorateCookedElement((element, helper) => {
+    const counterWrapper = helper.renderGlimmer(
+      "div.my-counter",
+      WelcomeBanner
+    );
+    element.appendChild(counterWrapper);
+  });
+});
+```
+
+`helper.getPost()` will return the current post, and can be used to build conditional logic into these `decorateCookedElement` callbacks. `console.log` the post to see what's available.
+
+### registerValueTransformer()
+
+`api.registerValueTransformer` allows you to inject logic into predefined parts of the Discourse JavaScript application. For example, you can add a `"home-logo-href"` transformer to link the logo to `example.com`:
+
+```
+api.registerValueTransformer("home-logo-href", () => "https://example.com")
+```
+
+For more information on Transformers, check out the [dedicated guide](https://meta.discourse.org/t/349954)
+
+## Finding more JS API methods
+
+All the available APIs are listed in the [`plugin-api.gjs` source code](https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/app/lib/plugin-api.gjs) in Discourse core, along with a short description and examples.
+
+That's it for this chapter, and almost the end of the tutorial. Let's wrap things up [in the conclusion]().
```

Comment:
Link will be added in a followup PR, because we don't have a topic-id for this doc yet. (🐔 / 🥚 problem)